### PR TITLE
GitHub Actions CI

### PR DESCRIPTION
**Pull request by pshriwise**
This adds a GitHub actions workflow for CI, replicating what we have in Travis. I was pleased with how easy it was to setup. Most of the Travis system was transferrable, requiring only a few syntax changes.

A couple of notable changes here:
  
  - An update from ubuntu 16.04 to 18.04 in CI. (Multiple ubuntu versions will be supported in the future.)
  - A bug fix in the C++ driver regression test for build with MPI.

There are further improvements to make, but this should allow us to continue development while we work on improving the CI/CD in GHA. I'll add intended improvements to #1731 for further discussion.

### CODE REVIEW

**Comment by Shimwell** [comment]
I noticed the checks still include Travis. Should this be removed from the checks that run

Screenshot from 2020-12-14 23-33-40

**Review by Shimwell — APPROVED** [review]
Looks great to me, nice to have all the CI all together with the code. Although the badge currently reports no status I think that is due to merging from a fork. I guess that will go green after the CI runs on the push trigger

Screenshot from 2020-12-15 15-03-21

[GitHub Actions build status (Linux)](https://github.com/openmc-dev/openmc/actions)

**Comment by pshriwise** [comment]
> I noticed the checks still include Travis. Should this be removed in the future

@Shimwell For sure. We'll want to deactivate our Travis build, remove the YAML file, and rename our CI scripts. I'm intending on that begin part of the improvements PR next.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      -
+        name: Environment Variables
+        run: |
+          echo "OPENMC_CROSS_SECTIONS=$HOME/nndc_hdf5/cross_sections.xml" >> $GITHUB_ENV
+          echo "OPENMC_ENDF_DATA=$HOME/endf-b-vii.1" >> $GITHUB_ENV
```
Comment (paulromano) [inline .github/workflows/ci.yml:70]: Shouldn't these be picked up from the `env:` block above?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        name: test
+        shell: bash
+        run: $GITHUB_WORKSPACE/tools/ci/travis-script.sh
+
+      - 
+        name: after_success
+        shell: bash
+        run: |
+          cpp-coveralls -i src -i include --exclude-pattern "/usr/*" --dump cpp_cov.json
+          coveralls --merge=cpp_cov.json
```
Comment (paulromano) [inline .github/workflows/ci.yml:109]: With our setup before, Travis CI relied on a webhook to indicate to coveralls that the job was done and it should report back the coverage. To achieve the same thing with GHA, it looks like we need an [extra job](https://github.com/coverallsapp/github-action#complete-parallel-job-example) that depends on the main job:
```
  finish:
    needs: main
    runs-on: ubuntu-latest
    steps:
    - name: Coveralls Finished
      uses: coverallsapp/github-action@master
      with:
        github-token: ${{ secrets.github_token }}
        parallel-finished: true
```
Hopefully that should result in the coverage getting reported back as an additional status check.

File: tools/ci/travis-install.sh [hunk 3]
```diff
@@ -1,6 +1,11 @@
 #!/bin/bash
 set -ex
 
+# Upgrade pip, pytest, numpy before doing anything else
+pip install --upgrade pip
+pip install --upgrade pytest
+pip install --upgrade numpy
```
Comment (paulromano) [inline tools/ci/travis-install.sh:7]: Why did this block need to be moved up?

File: README.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # OpenMC Monte Carlo Particle Transport Code
 
 [![License](https://img.shields.io/github/license/openmc-dev/openmc.svg)](http://openmc.readthedocs.io/en/latest/license.html)
-[![Travis CI build status (Linux)](https://travis-ci.org/openmc-dev/openmc.svg?branch=develop)](https://travis-ci.org/openmc-dev/openmc)
+[![GitHub Actions build status (Linux)](https://github.com/pshriwise/openmc/workflows/CI/badge.svg?branch=develop)](https://github.com/openmc-dev/openmc/actions)
```
Comment (paulromano) [inline README.md:4]:
```suggestion
[![GitHub Actions build status (Linux)](https://github.com/openmc-dev/openmc/workflows/CI/badge.svg?branch=develop)](https://github.com/openmc-dev/openmc/actions)
```

**Review by paulromano — CHANGES_REQUESTED** [review]
Thanks for setting this up @pshriwise! The major thing that still needs fixing is reporting of coverage, but I have a suggestion below that I think should do the trick. Also, please go ahead and remove the .travis.yml file; otherwise the PR is still expecting Travis to report back and nobody can merge. If you want to rename the scripts too, I'd say go ahead.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (pshriwise) [reply .github/workflows/ci.yml:70]: Ah, yeah they should. Removing...

File: tools/ci/travis-install.sh (hunk 3 above)
Comment (pshriwise) [reply tools/ci/travis-install.sh:7]: It seems that some of the pip peripherals aren't available by default on GHA images. In this case, `wheels` isn't added until we upgrade pip. This causes problems in the `vectfit` build.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (pshriwise) [reply .github/workflows/ci.yml:112]: Added this in. I'll check coveralls when CI completes to make sure it updated properly.

Comment (pshriwise) [reply .github/workflows/ci.yml:112]: Looks like it uploaded: https://coveralls.io/builds/35772557 :heavy_check_mark:

**Review by paulromano — APPROVED** [review]
Thanks again @pshriwise! Hooray for working CI :tada:

**Pull request merged**